2928 Refactor lesmis vitals views

tcaiger opened the pull request:
### Issue #: [2928](https://github.com/beyondessential/tupaia-backlog/issues/2928) refactor lesmis dashboard view

### Changes:
- Split vitals templates into a new file called VitalsViews
- Update vitals views to match design
- Improve loading states on vitals

Note: there is still some issues with data fetching logic in the fetch handling for vitals but that is out of scope of this ticket. See [other ticket](https://app.zenhub.com/workspaces/active-sprints-5eea9d3de8519e0019186490/issues/beyondessential/tupaia-backlog/2772)

---

tcaiger commented on packages/lesmis/src/api/queries/useVitalsData.js:
```diff
@@ -166,13 +166,13 @@ export const useVitalsData = entityCode => {
-    ...districtData,
-    ...provinceData,
-
+    data: {
```
add api data to the data key to keep the react-query convention

tcaiger commented on packages/lesmis/src/components/EntityVitalsItem.js:
```diff
@@ -80,34 +54,57 @@ const VitalsIcon = ({ icon }) => {
-  }
-  return <VitalContent>{value || '-'}</VitalContent>;
-};
+const VitalName = styled(Typography)`
```
update styles to match design

tcaiger commented on packages/lesmis/src/views/DashboardReportTabView.js:
```diff
@@ -166,7 +166,7 @@ export const DashboardReportTabView = ({
             subDashboards.map(dashboard => (
               <TabPanel
-                key={dashboard.dashboardName}
+                key={dashboard.dashboardId}
```
update to use a unique key

tcaiger commented on packages/lesmis/src/views/DashboardView.js:
```diff
@@ -463,69 +163,52 @@ const useDefaultDashboardTab = (selectedDashboard = null, options) => {
+                  <YearSelector
+                    value={selectedYear}
+                    onChange={setSelectedYear}
+                    isLoading={!!isFetching}
```
Set up loading state on year selector. This will be used in the next PR.

tcaiger commented on packages/lesmis/src/views/VitalsViews.js:
```diff
@@ -0,0 +1,357 @@
+      <EntityVitalsItem name="Province Code" value={vitals.code} icon="LocationPin" />
+      <EntityVitalsItem
+        name="Province Population"
+        value={vitals.Population?.toLocaleString()}
```
Add commas to numbered fields

edmofro commented on packages/lesmis/src/views/VitalsViews.js:
```diff
@@ -0,0 +1,357 @@
+        value={vitals.NumberOfStudents?.toLocaleString()}
+        icon="Study"
+      />
+      <EntityVitalsItem name="Complete School" icon="Notepad" isLoading={vitals.isLoading} />
```
```suggestion
      <EntityVitalsItem name="Complete School" icon="Notepad" />
```

edmofro commented on packages/lesmis/src/views/DashboardView.js:
```diff
@@ -61,394 +55,100 @@ const getProfileLabel = entityType => {
-    ComponentProps: {
-      filterSubDashboards: subDashboardFilters.essdpEarlyChildhood,
+    TabComponent: DashboardReportTabView,
+    useYearSelector: true,
```
```suggestion
```

edmofro commented on packages/lesmis/src/views/VitalsViews.js:
```diff
@@ -0,0 +1,357 @@
+  height: 100%;
+`;
+
+const PartnersSection = ({ vitals, displayAll }) => {
```
I reckon this PartnersSection could be split out again

edmofro reviewed (changes requested):
Somewhat inclined to split up the vitals file further into:
- partners section
- each entity type vital
- the loader + vitals outer component

But totally up to you. Only required changes are a couple of very small ones inline

tcaiger commented:
@edmofro - I have added the small inline changes and done a bit more refactoring based on your nudging to split up the file further and based on always showing the partners section. Here is a summary:

- move the data processing for the partners logos into `useEntityVitals` and moved partners section template inline
- moved the map/photo components inline
- - extracted out the Vitals loader into a new file
- renamed VitalsViews to VitalsView

I decided to keep the Vitals entity templates in the same file as they are super specific to the VitalsView and use the vitals props which could be confusing out of context.

edmofro commented on packages/lesmis/src/api/queries/useVitalsData.js:
```diff
@@ -165,14 +196,18 @@ export const useVitalsData = entityCode => {
-      ...districtData,
-      ...provinceData,
-    },
+    data: { ...vitalsData, partners },
```
Great idea to move the data to here!!

edmofro reviewed (approved):
Yep all looks good :-)
